src/GameSession.js

```diff
@@ -1,44 +1,101 @@
-const Squisher = require("./Squisher");
+const Squisher = require('./Squisher');
 const { generateName } = require('./common/util/name-generator');
 
 class GameSession {
-    constructor(game, res) {
+    constructor(game) {
         this.game = game;
-        this.squisher = new Squisher(res.width, res.height, game);
+        this.squisher = new Squisher(this.game);
         this.squisher.addListener(this);
-        this.players = {};
-        this.game.players = this.players;
-    }
-
-    async addPlayer(player) {
-        player.name = await generateName();
-         
-        player.addInputListener(this.squisher);
-        player.addStateListener(this);
-        this.players[player.id] = player;
-        const gameAssets = await this.squisher.getAssets();
-        if (gameAssets && gameAssets.length > 0) {
-            player.receiveUpdate(gameAssets);
+        this.renderWidth = this.game.constructor.metadata ? this.game.constructor.metadata().res.width : 1280;
+        this.renderHeight = this.game.constructor.metadata ? this.game.constructor.metadata().res.height : 720;
+    }
+
+    handleSquisherUpdate(squished) {
+        for (const playerId in this.game.players) {
+            this.game.players[playerId].receiveUpdate(squished);
+        }
+    }
+
+    addPlayer(player) {
+        generateName().then(playerName => {
+            player.name = playerName;
+            this.squisher.assetBundle && player.receiveUpdate(this.squisher.assetBundle);
+
+            player.receiveUpdate(this.squisher.squished);
+            this.game.addPlayer(player);
+            this.game.handleNewPlayer && this.game.handleNewPlayer(player);
+            player.addInputListener(this);
+        });
+
+    }
+
+    handlePlayerDisconnect(playerId) {
+        this.game.handlePlayerDisconnect && this.game.handlePlayerDisconnect(playerId);
+        this.game.removePlayer(playerId);
+    }
+
+    initialize(cb) {
+        if (this.initialized) {
+            cb && cb();
+        } else {
+            this.squisher.initialize().then(() => {
+                this.initialized = true;
+                cb && cb();
+            });
         }
-        player.receiveUpdate(this.squisher.getPixels());
+    }
 
-        this.game.handleNewPlayer && this.game.handleNewPlayer(player);
+    handlePlayerInput(player, input) {
+        if (input.type === 'click') {
+            this.handleClick(player, input.data);
+        } else if (input.type === 'keydown') { 
+            this.game.handleKeyDown && this.game.handleKeyDown(player, input.key);
+        } else if (input.type === 'keyup') {
+            this.game.handleKeyUp && this.game.handleKeyUp(player, input.key);
+        } else {
+            console.log('Unknown input type: ' + input.type);
+        }
     }
 
-    getGameState() {
-        return this.squisher.getPixels();
+
+    handleClick(player, click) {
```

Comment:
I don't think this stuff should be in the game session. I think I want a generic input mapper module (maybe something like what we have for squishjs, something that can be shared on the client and server)

But that's stuff we can add later. As noted in my description for this PR, I don't want to keep adding since I'm the king of feature creep and I'm trying to avoid that